### DISCUSSION TITLE
Raspberry Pi - SSD

### DISCUSSION THREAD

**Discussion by SanderVreeken** [coollabsio/coolify#2522] [Q&A]
Yesterday I started using Coolify via my Raspberry Pi and all works great. Currently it’s running via an SD card. Unfortunately this has only limited space available and does not read and write fast enough.

How can I best use the SSD connected to the Pi? Should I use the SDD to boot the Pi and Coolify instead? Or is there a better alternative? E.g., setting in the dashboard.

**Comment by ashikpy** [comment]
yes use m.2 hat (you can buy online) and use it, for loading and install it there!

by the way can you help me with installing coolify? my discord - @http.ash.in

**Comment by Lintshiwe** [comment]
## Recommended Approach: Saga Pattern

For a car wash microservices architecture, the **Saga pattern** is the most practical choice:

### Why Saga?
1. **No distributed locks** - Unlike 2PC, Sagas avoid blocking
2. **Compensation support** - If payment fails, booking can be cancelled
3. **Eventual consistency** - Each service manages its own data

### Implementation:
- Use choreography-based Saga with events
- Booking service publishes `BookingCreated` event
- Payment service listens and processes
- If payment fails, publish `PaymentFailed` event
- Booking service compensates by cancelling

### Tools:
- Spring Cloud Stream for event messaging
- RabbitMQ or Kafka as message broker
- Idempotent consumers for reliability

This approach has proven effective in production for similar booking systems.
